### ISSUE TITLE
Packages add/remove with --dev does not work if custom scripts configuration block defined [tool.pdm.scripts]

### ISSUE THREAD

**Issue by mrx88** [pdm-project/pdm#2056]
- [x ] I have searched the issue tracker and believe that this is not a duplicate.

## Steps to reproduce

1. Add tool.pdm.scripts to pyproject.toml:
`[tool.pdm.scripts]
hello = {shell = "echo hello"}`
2. Add or Remove package from dev, e.g pytest:
`pdm add pytest --dev`
3. Package not added/removed under [tool.pdm.dev-dependencies]  "dev" section. Package also not installed or removed.

Note: if removing [tool.pdm.scripts] section, step 2 works fine.

## Actual behavior

--dev packages add/remove does not work if custom scripts configuration block defined [tool.pdm.scripts]

## Expected behavior

--dev packages add/remove should work if custom scripts defined under [tool.pdm.scripts]

## Environment Information

```bash
# Paste the output of `pdm info && pdm info --env` below:

PDM version:
  5.7.4
Python Interpreter:
  /home/asd/azdevops/apps/asd/.venv/bin/python (3.9)
Project Root:
  /home/asd/azdevops/apps/asd
Local Packages:
  
{
  "implementation_name": "cpython",
  "implementation_version": "3.9.2",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "5.15.90.1-microsoft-standard-WSL2",
  "platform_system": "Linux",
  "platform_version": "#1 SMP Fri Jan 27 02:56:13 UTC 2023",
  "python_full_version": "3.9.2",
  "platform_python_implementation": "CPython",
  "python_version": "3.9",
  "sys_platform": "linux"
}
```

**Comment by frostming** [comment]
This bug report makes no sense to me. I can't reproduce it either. There should be some problem on your side, can you provide a example repository(or docker image) to reproduce this issue?

**Comment by mrx88** [comment]
Yes it is strange one, but I have not figured it out yet, therefore created this issue to get more ideas. Strange that happens with my work laptop and I can`t reproduce it on my personal one. In the example I am using python 3.9.2 (which has issues, Win 10, WSL, VS Code) and in personal project python 3.11(no issues, Win 11, WSL, VS Code)

pdm list |grep pytest                                        

│ pytest             │ 7.4.0    │          │

pdm remove pytest --dev                                    
Removing packages from dev dev-dependencies: pytest
Changes are written to pyproject.toml.
🔒 Lock successful
Changes are written to pdm.lock.
All packages are synced to date, nothing to do.

🎉 All complete!

Package still in the list:
pdm list |grep pytest                             
│ pytest             │ 7.4.0    │          │

Also not removed from .toml conf under tool.pdm.dev-dependencies section.

venv path is correct.

**Comment by frostming** [comment]
@mrx88 This doesn't help at all. You may need to do some investigation yourself.

**Comment by mrx88** [comment]
I figured out how to fix it:

[tool.pdm.scripts] block was under [build-system] in toml config, when moving it above [build-system] package remove/add and toml config update works as expected. Not sure if such behavior is expected or it`s a bug.

**Comment by frostming** [comment]
So you mean such structure:

```toml

[tool.pdm.dev-dependencies]
dev = ["pytest"]

[build-system]
...

[tool.pdm.scripts]
...
```

Noted, will try it tomorrow.

**Comment by frostming** [comment]
Fixed by #2061
